packages/apollo-graphql/src/__tests__/signature.test.ts

```diff
@@ -106,6 +106,121 @@ describe('aggressive signature', () => {
       `,
       output: '{user{name...Bar}}fragment Bar on User{asd}',
     },
+    {
+      name: 'fragments out of order',
+      operationName: '',
+      input: gql`
+        fragment Baz on User {
+          jkl
+        }
+
+        {
+          user {
+            name
+            ...Bar
+            ...Baz
+          }
+        }
+
+        fragment Bar on User {
+          asd
+        }
+      `,
+      output:
+        '{user{name...Bar...Baz}}fragment Bar on User{asd}fragment Baz on User{jkl}',
+    },
+    {
+      name: 'fragments in order',
+      operationName: '',
+      input: gql`
+        {
+          user {
+            name
+            ...Bar
+            ...Baz
+          }
+        }
+
+        fragment Bar on User {
+          asd
+        }
+
+        fragment Baz on User {
+          jkl
+        }
+      `,
+      output:
+        '{user{name...Bar...Baz}}fragment Bar on User{asd}fragment Baz on User{jkl}',
+    },
+    {
+      name: 'fragments in order',
+      operationName: '',
+      input: gql`
+        {
+          user {
+            name
+            ...Bar
+            ...Baz
+          }
+        }
+
+        fragment Bar on User {
+          asd
+        }
+
+        fragment Baz on User {
+          jkl
+        }
+      `,
+      output:
+        '{user{name...Bar...Baz}}fragment Bar on User{asd}fragment Baz on User{jkl}',
+    },
+    {
+      name: 'with a subscription',
+      operationName: '',
+      input: gql`
+        fragment Bar on User {
+          asd
+        }
+
+        fragment Baz on User {
+          jkl
+        }
+
+        subscription A {
+          user {
+            name
+            ...Bar
+            ...Baz
+          }
+        }
+      `,
+      output:
+        'subscription A{user{name...Bar...Baz}}' +
+        'fragment Bar on User{asd}fragment Baz on User{jkl}',
+    },
+    {
+      name: 'with a mutation',
+      operationName: '',
+      input: gql`
+        mutation A {
+          user {
+            name
+            ...Bar
+            ...Baz
+          }
+        }
+        fragment Baz on User {
+          jkl
+        }
+        fragment Bar on User {
+          asd
+        }
+      `,
+      output:
+        'mutation A{user{name...Bar...Baz}}' +
+        'fragment Bar on User{asd}fragment Baz on User{jkl}',
+    },
```

Comment:
This is very helpful articulation of some behavior I was seeing during further implementation today.

I'll gladly add some tests specific to what you've mentioned (if I haven't already - will go back and take a look). I went a little crazy adding tests since I found TDD to be the fastest and best way to implement these changes :woo:!